libs/watt/package/picker/datepicker/watt-datepicker.component.ts

```diff
@@ -100,71 +100,26 @@ export class WattDatepickerComponent extends WattPickerBase implements Validator
   dateClass = input<MatCalendarCellClassFunction<Date>>(() => '');
   canStepThroughDays = input(false, { transform: booleanAttribute });
 
-  /**
-   * @ignore
-   */
-  @ViewChild(MatDatepickerInput)
-  matDatepickerInput!: MatDatepickerInput<Date | null>;
-
-  @ViewChild(MatDateRangePicker)
-  matDateRangePicker!: MatDateRangePicker<Date | null>;
-
-  @ViewChild(MatDateRangeInput)
-  matDateRangeInput!: MatDateRangeInput<Date | null>;
-
-  /**
-   * @ignore
-   */
-  @ViewChild(MatStartDate)
-  matStartDate!: MatStartDate<Date | null>;
+  protected matDatepickerInput =
+    viewChild.required<MatDatepickerInput<Date | null>>(MatDatepickerInput);
+  protected matDateRangePicker =
+    viewChild.required<MatDateRangePicker<Date | null>>(MatDateRangePicker);
+  protected matDateRangeInput =
+    viewChild.required<MatDateRangeInput<Date | null>>(MatDateRangeInput);
 
-  /**
-   * @ignore
-   */
-  @ViewChild(MatEndDate)
-  matEndDate!: MatEndDate<Date | null>;
+  protected matStartDate = viewChild.required<MatStartDate<Date | null>>(MatStartDate);
+  protected matEndDate = viewChild.required<MatEndDate<Date | null>>(MatEndDate);
+  protected actualInput = viewChild.required<ElementRef<HTMLInputElement>>('actualInput');
```

Comment:
issue: `viewChild.required` will throw an error at run-time if the token can't be found. And since the inputs are inside `if/else` not all of them will be available all the time. So some of these `viewChild` queries will fail.